### PR TITLE
Drive selector button

### PR DESCRIPTION
**Pull request by amdomanska**
Convert angular component to React using Rendition library

### CODE REVIEW

**Comment by amdomanska** [comment]
@resinci retest

File: lib/gui/app/components/drive-selector-button/drive-selector-button.jsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  reselectDrive: propTypes.func,
+  driveSize: propTypes.string,
+  hasCompatibilityStatus: propTypes.func,
+  getCompatibilityStatuses: propTypes.func,
+  drives: propTypes.func,
+  image: propTypes.func,
+  selectedDevices: propTypes.array,
+}
+
+module.exports = DriveSelectorButton
```
Comment (LucianBuzzo) [inline lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: Can you use a named export here?

File: lib/gui/app/components/drive-selector-button/drive-selector-button.jsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      return this.props.selectedDevices.map((device) =>
+        <Txt key={device.device} tooltip={device.description + '(' + device.displayName + ')'}>
+          { middleEllipsis(device.description, 14) }
+        </Txt>
+       )
+    }
+  }
+
+  render() {
+    if (this.props.hasDrive || !this.props.shouldShowDrivesButton) {
```
[the remark this answers is not in the document]
Comment (amdomanska) [reply lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: It's the property set in settings, it is decided outside component.

File: lib/gui/app/components/drive-selector-button/drive-selector-button.jsx (hunk 1 above)
Comment (amdomanska) [reply lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: I cannot as long as we need to use React2Angular component - I haven't found the way for it to accept the named export. Once the whole main page will be converted to React this is going to be changed.

Comment (LucianBuzzo) [reply lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: @amdomanska How are you importing this module? modules and `require` statements should have nothing to do with `react2angular`.

**Comment by thundron** [comment]
@resin-ci retest

**Comment by thundron** [comment]
@resin-ci retest

File: lib/gui/app/components/drive-selector-button/drive-selector-button.jsx (hunk 1 above)
Comment (thundron) [reply lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: Tests fail for some reason when we use named exports, I still need to take a closer look at this

File: lib/gui/app/components/drive-selector-button/drive-selector-button.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (thundron) [reply lib/gui/app/components/drive-selector-button/drive-selector-button.jsx]: Confirmed, it's in the angular side for now. When we transition all of this logic to React, we'll be able to change names too

**Pull request closed without merging**